rbac/management/workspace/model.py

```diff
@@ -27,17 +28,35 @@
 class Workspace(TenantAwareModel):
     """A workspace."""
 
+    class Types(models.TextChoices):
+        STANDARD = "standard"
+        DEFAULT = "default"
+        ROOT = "root"
+
     name = models.CharField(max_length=255)
     uuid = models.UUIDField(default=uuid4, editable=False, unique=True, null=False)
     parent = models.ForeignKey(
         "self", to_field="uuid", on_delete=models.PROTECT, related_name="children", null=True, blank=True
     )
     description = models.CharField(max_length=255, null=True, blank=True, editable=True)
+    type = models.CharField(choices=Types.choices, default=Types.STANDARD, null=False)
     created = models.DateTimeField(default=timezone.now)
     modified = AutoDateTimeField(default=timezone.now)
 
     class Meta:
         ordering = ["name", "modified"]
+        constraints = [
+            UniqueConstraint(
+                fields=["tenant_id", "type"],
+                name="unique_default_root_workspace_per_tenant",
+                condition=Q(type__in=["root", "default"]),
+            )
+        ]
+
+    def save(self, *args, **kwargs):
+        """Override save on model to enforce validations."""
+        self.full_clean()
```

Comment:
Curious what happens if we don't add this one? Will the save pass the constrain validation?

Reply:
The uniqueness constraints would still trigger, but the type choices would not be enforced, so you could create a workspace of type "foo".